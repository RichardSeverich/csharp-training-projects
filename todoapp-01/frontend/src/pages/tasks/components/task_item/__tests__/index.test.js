import React from 'react';
import TaskItem from '../.';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
// eslint-disable-next-line jest/no-mocks-import
import taskMock from '../__mocks__/task';
import AppProvider from 'context/application/app';

/**
 * @param {object} target mocked onClick function
 * @returns {object[]} amount of times the event was triggered.
 */
function getCalls(target) {
	return target.mock.calls;
}

describe('pages/tasks/components/task_item', () => {
	describe('Html', () => {
		it('Should render', () => {
			const { container } = render(<TaskItem value={{ ...taskMock }} />);
			expect(container.firstChild).toBeInstanceOf(HTMLDivElement);
		});

		it('Should render correctly', () => {
			const { container } = render(<TaskItem value={{ ...taskMock }} />);
			expect(container.firstChild).toMatchSnapshot();
		});
	});

	describe('Interactions', () => {
		it('Should trigger event onClick', () => {
			const mockOnClick = jest.fn();
			const { getByTitle } = render(
				<AppProvider>
					<Router>
						<TaskItem value={{ ...taskMock }} onDelete={mockOnClick} />
					</Router>
				</AppProvider>
			);
			let button = getByTitle('Delete task');
			fireEvent.click(button);
			expect(getCalls(mockOnClick).length).toBeGreaterThanOrEqual(1);
		});
	});
});
