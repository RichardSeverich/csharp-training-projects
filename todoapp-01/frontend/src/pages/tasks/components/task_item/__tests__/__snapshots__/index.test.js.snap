// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/tasks/components/task_item Html Should render correctly 1`] = `
<div
  class="ui card Low"
  id="1"
>
  <div
    class="content"
  >
    <div
      class="right aligned meta"
    >
      <i
        aria-hidden="true"
        class="calendar plus icon"
      />
       Due: 
      No due date
    </div>
    <div
      class="header"
    >
      Research about imaginary numbers
    </div>
    <div
      class="description"
    >
      Task 
      Completed
    </div>
    <div
      class="left aligned meta"
    >
      <i
        aria-hidden="true"
        class="calendar plus icon"
      />
       Created on: 
      12/12/2020
    </div>
    <br />
    <div
      class="meta"
    >
      No tags present
    </div>
  </div>
  <div
    class="extra content"
  >
    <div
      class="ui two buttons"
    >
      <button
        class="ui red inverted button"
        title="Delete task"
      >
        <i
          aria-hidden="true"
          class="trash icon"
        />
      </button>
    </div>
  </div>
</div>
`;
